lib/galaxy/tools/parameters/basic.py

```diff
@@ -235,6 +234,17 @@ def build( cls, tool, param ):
         else:
             return parameter_types[param_type]( tool, param )
 
+    @classmethod
+    def parse_name(cls, input_source):
+        name = input_source.get("name", None)
+        if name is None:
+            argument = input_source.get("argument")
+            if argument:
+                name = argument.lstrip("-")
+            else:
```

Comment:
It shouldn't modify argument - to me it makes sense that if you specify the argument as ``--input`` - the name can safely be inferred to be ``input``. This provides some actual convenience to tool authors annotating things this way - argument will still be ``--input``.

Put another way...

```
<param name="input" argument="--input"... 
```

feels redundant right?